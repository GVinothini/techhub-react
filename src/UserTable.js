import { Component } from "react";

export default class UserTable extends Component {
    render() {
        const userList = this.props.userList;
        const userListTemplate = [];
        for (let i = 0; i < userList.length; i++) {
            userListTemplate.push(
                <tr key={`${i}${userList[i]}`}>
                    <th scope="row">{i}</th>
                    <td>{userList[i]}</td>
                </tr>
            );
        }
        return (
            <table className="table table-condensed">
                <thead>
                    <tr>
                        <th>#</th>
                        <th>First Name</th>
                    </tr>
                </thead>
                <tbody>{userListTemplate}</tbody>
            </table>
        );
    }
}
