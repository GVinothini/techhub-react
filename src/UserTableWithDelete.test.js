import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import UserTable from "./UserTableWithDelete";

global.alert = jest.fn();

describe('UserTable', () => {
    const mockProps = {
        userList: [ 'test1', 'test2'],
        deleteUser: jest.fn()
    };

    it('should render UserTable', () => {
       const { container } = render(<UserTable {...mockProps} />);
       expect(container).toMatchSnapshot();
    });

    it('should call deleteUser while click save button', () => {
        render(<UserTable {...mockProps} />);
        const [firstDeleteButton] = screen.getAllByText('Delete');

        // Fires the Click event to verify the alert when the row is in 'ReadOnly' mode.
        fireEvent.click(firstDeleteButton);
        expect(global.alert).toBeCalled();

        // Getting 'Make ReadWrite' button Element from first row(user).
        // Fires the click event to change the button text to 'Make ReadOnly'
        const [firstButton] = screen.getAllByText('Make ReadWrite');
        fireEvent.click(firstButton);

        // Getting 'Make ReadOnly' button Element from first row(user).
        // Verifying the button text is changed successfully.
        const [changedButton] = screen.getAllByText('Make ReadOnly');
        expect(changedButton).toBeInTheDocument();

        // Fires the Click event to delete the user when the row is in 'ReadWrite' mode.
        fireEvent.click(firstDeleteButton);
        expect(mockProps.deleteUser).toBeCalled();
        
    });
    
});
