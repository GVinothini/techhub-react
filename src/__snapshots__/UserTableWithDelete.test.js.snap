// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTable should render UserTable 1`] = `
<div>
  <table
    class="table table-condensed"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          First Name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          0
        </th>
        <td>
          test1
        </td>
        <td>
          <button
            type="button"
          >
            Make ReadWrite
          </button>
        </td>
        <td>
          <button
            type="button"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          1
        </th>
        <td>
          test2
        </td>
        <td>
          <button
            type="button"
          >
            Make ReadWrite
          </button>
        </td>
        <td>
          <button
            type="button"
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
